commons-geometry-core/src/main/java/org/apache/commons/geometry/core/partitioning/bsp/AbstractBSPTree.java

```diff
@@ -343,7 +343,7 @@ protected N findNode(final N start, final P pt, final FindNodeCutRule cutRule) {
             final boolean onMinusSide = cutLoc == HyperplaneLocation.MINUS;
             final boolean onCut = !onPlusSide && !onMinusSide;
 
-            if (onMinusSide || (onCut && cutRule == FindNodeCutRule.MINUS)) {
+            if (onMinusSide || onCut && cutRule == FindNodeCutRule.MINUS) {
```

Comment:
I agree that the parentheses here increase readability. Unfortunately the checkstyle rule is not configurable: [UnnecessaryParentheses](https://checkstyle.sourceforge.io/config_coding.html#UnnecessaryParentheses)

It would be nice to be able to turn it off for expressions involving `&&` or `||` but leave it on for the other checks with parentheses. In this case I would add a suppression for the class to allow this. You can add a line number if you want to be specific.